File: leetcode/medium/1055-shortest-way-to-form-string/1055-shortest-way-to-form-string.kt
class Solution {
    fun shortestWay(source: String, target: String): Int {
        var repeatCount = 0
        var sourceIndex = 0
        val sourceLength = source.length

        for (i in target.indices) {
            val targetChar = target[i]
            var found = false

            for (j in sourceIndex until (sourceIndex + sourceLength)) {
                if (j % sourceLength == 0) {
                    repeatCount++
                }

                if (source[j % sourceLength] == targetChar) {
                    found = true
                    sourceIndex = j + 1
                    break
                }
            }

            if (!found) {
                return -1
            }
        }

        return repeatCount
    }
}

File: leetcode/medium/1055-shortest-way-to-form-string/1055-shortest-way-to-form-string.py
class Solution:
    def shortestWay(self, source: str, target: str) -> int:
        repeatCount = 0
        sourceIndex = 0
        sourceLength = len(source)

        for targetChar in target:
            found = False

            for i in range(sourceIndex, sourceIndex + sourceLength):
                if (i % sourceLength) == 0:
                    repeatCount += 1

                if source[i % sourceLength] == targetChar:
                    sourceIndex = i + 1
                    found = True
                    break

            if not found:
                return -1

        return repeatCount
